devito/ir/clusters/algorithms.py

```diff
@@ -106,8 +106,43 @@ def guard(clusters):
 
             guards = [(i.parent, conditions[i]) for i in dims]
             guards.extend([(i.parent, negated[i]) for i in ndims])
-            cluster = PartialCluster(exprs, c.ispace, c.dspace, c.atomics, dict(guards))
-            processed.append(cluster)
+            guards = dict(guards)
+
+            # Merge of conditonals if both conditions. In case of multiple Conditionals
+            # we can have a case where we have:
```

Comment:
The following block of code is quite messy. I need to take a look at it. Next week.

Reply:
I'm not sure, the kill of some branch does not depend only on the dimension but on the expression too that the poewrset just ignores.